selfie.c

```diff
@@ -2291,27 +2293,62 @@ uint64_t isCharacterWhitespace() {
 }
 
 uint64_t findNextCharacter() {
-  uint64_t inComment;
+  uint64_t singleLineComment;
+  uint64_t multiLineComment;
 
   // assuming we are not in a comment
-  inComment = 0;
+  singleLineComment = 0;
+  multiLineComment = 0;
 
   // read and discard all whitespace and comments until a character is found
   // that is not whitespace and does not occur in a comment, or the file ends
   while (1) {
-    if (inComment) {
+    if (singleLineComment) {
       getCharacter();
 
       if (isCharacterNewLine())
         // comments end with new line
-        inComment = 0;
+        singleLineComment = 0;
       else if (character == CHAR_EOF)
         return character;
       else
         // count the characters in comments as ignored characters
         // line feed and carriage return are counted below
         numberOfIgnoredCharacters = numberOfIgnoredCharacters + 1;
 
+    } else if (multiLineComment) {
+      getCharacter();
+
+      // comments end with */
+      if (character == CHAR_ASTERISK) {
+      	getCharacter();
+
+      	if (character == CHAR_SLASH) {
+      	  multiLineComment = 0;
+         
+         // count */ as two ignored characters
+         numberOfIgnoredCharacters = numberOfIgnoredCharacters + 2;
+
+      	  getCharacter(); // next character after */
+      	} else if (character == CHAR_EOF)
```

Comment:
Can you try to avoid duplicating the code here? Check for whitespace before checking for `inMultiLineComment`.